modules/local-caching-classloader/src/test/java/org/apache/accumulo/classloader/lcc/util/LocalStoreTest.java

```diff
@@ -0,0 +1,259 @@
+      jetty.stop();
+      jetty.join();
+    }
+    if (hdfs != null) {
+      hdfs.shutdown();
+    }
+  }
+
+  @AfterEach
+  public void cleanBaseDir() throws Exception {
+    if (Files.exists(baseCacheDir)) {
+      try (var walker = Files.walk(baseCacheDir)) {
+        walker.map(java.nio.file.Path::toFile).sorted(Comparator.reverseOrder())
+            .forEach(File::delete);
+      }
+    }
+  }
+
+  @Test
+  public void testPropertyNotSet() {
+    assertThrows(NullPointerException.class, () -> new LocalStore(null));
+  }
+
+  @Test
+  public void testCreateBaseDirs() throws Exception {
+    assertFalse(Files.exists(baseCacheDir));
+    var localStore = new LocalStore(baseCacheDir);
+    assertTrue(Files.exists(baseCacheDir));
+    assertTrue(Files.exists(baseCacheDir.resolve("contexts")));
+    assertTrue(Files.exists(baseCacheDir.resolve("resources")));
+    assertEquals(baseCacheDir.resolve("contexts"), localStore.contextsDir());
+    assertEquals(baseCacheDir.resolve("resources"), localStore.resourcesDir());
+  }
+
+  @Test
+  public void testCreateBaseDirsMultipleTimes() throws Exception {
+    assertFalse(Files.exists(baseCacheDir));
+    assertNotNull(new LocalStore(baseCacheDir));
+    assertNotNull(new LocalStore(baseCacheDir));
+    assertNotNull(new LocalStore(baseCacheDir));
+    assertNotNull(new LocalStore(baseCacheDir));
+    assertTrue(Files.exists(baseCacheDir));
+  }
+
+  @Test
+  public void testLocalFileName() {
+    // regular json
+    assertEquals("f1-chk1.json", LocalStore.localName("f1.json", "chk1"));
+    // dotfile json
+    assertEquals(".f1-chk1.json", LocalStore.localName(".f1.json", "chk1"));
+    // regular jar (has multiple dots)
+    assertEquals("f2-1.0-chk2.jar", LocalStore.localName("f2-1.0.jar", "chk2"));
+    // dotfile jar (has multiple dots)
+    assertEquals(".f2-1.0-chk2.jar", LocalStore.localName(".f2-1.0.jar", "chk2"));
+    // regular file with no suffix
+    assertEquals("f3-chk3", LocalStore.localName("f3", "chk3"));
+
+    // weird files with trailing dots and no file suffix
+    assertEquals("f4.-chk4", LocalStore.localName("f4.", "chk4"));
+    assertEquals("f4..-chk4", LocalStore.localName("f4..", "chk4"));
+    assertEquals("f4...-chk4", LocalStore.localName("f4...", "chk4"));
+    // weird dotfiles that don't really have a suffix
+    assertEquals(".f5-chk5", LocalStore.localName(".f5", "chk5"));
+    assertEquals("..f5-chk5", LocalStore.localName("..f5", "chk5"));
+    // weird files with weird dots, but do have a valid suffix
+    assertEquals("f6.-chk6.jar", LocalStore.localName("f6..jar", "chk6"));
+    assertEquals("f6..-chk6.jar", LocalStore.localName("f6...jar", "chk6"));
+    assertEquals(".f6-chk6.jar", LocalStore.localName(".f6.jar", "chk6"));
+    assertEquals("..f6-chk6.jar", LocalStore.localName("..f6.jar", "chk6"));
+    assertEquals(".f6.-chk6.jar", LocalStore.localName(".f6..jar", "chk6"));
+    assertEquals("..f6.-chk6.jar", LocalStore.localName("..f6..jar", "chk6"));
```

Comment:
There is code in `LocalStore.storeContextDefinition` that replaces the File separator character with an underscore. I wonder if that replacement should move into localFileName and add a test for it here.